engines/ml/xgboost/src/main/java/ai/djl/ml/xgboost/XgbNDArray.java

```diff
@@ -88,14 +88,36 @@ public ByteBuffer toByteBuffer(boolean tryDirect) {
     /** {@inheritDoc} */
     @Override
     public void intern(NDArray replaced) {
-        if (handle != null && handle.get() != 0L) {
-            long pointer = handle.getAndSet(0L);
-            JniUtils.deleteDMatrix(pointer);
+        if (replaced == null) {
+            throw new IllegalArgumentException("The replaced NDArray cannot be null.");
+        }
+        if (!(replaced instanceof XgbNDArray)) {
+            throw new IllegalArgumentException(
+                    "The replaced NDArray must be an instance of XgbNDArray.");
         }
         XgbNDArray array = (XgbNDArray) replaced;
-        data = array.data;
-        handle = array.handle;
-        format = array.format;
+
+        synchronized (this) {
+            if (handle != null && handle.get() != 0L) {
+                long pointer = handle.getAndSet(0L);
+                JniUtils.deleteDMatrix(pointer);
+            }
+
+            data = array.data;
+            format = array.format;
+
+            if (array.handle != null && array.handle.get() != 0L) {
```

Comment:
Not sure about this null check, if we want to prevent a closed NDArray, we should just do:
```
if (this.closed()
```